core/clients/blobberClient/client.go

```diff
@@ -270,3 +271,27 @@ func CalculateHash(url string, req *blobbergrpc.CalculateHashRequest) ([]byte, e
 
 	return json.Marshal(convert.GetCalculateHashResponseHandler(calculateHashResp))
 }
+
+func RenameObject(url string, req *blobbergrpc.RenameObjectRequest) ([]byte, error) {
+	blobberClient, err := newBlobberGRPCClient(url)
+	if err != nil {
+		return nil, err
+	}
+
+	clientSignature, err := client.Sign(encryption.Hash(req.Allocation))
+	if err != nil {
+		return nil, err
+	}
+
+	grpcCtx := metadata.NewOutgoingContext(context.Background(), metadata.New(map[string]string{
+		blobbercommon.ClientHeader:          client.GetClientID(),
+		blobbercommon.ClientKeyHeader:       client.GetClientPublicKey(),
+		blobbercommon.ClientSignatureHeader: clientSignature,
+	}))
+
+	renameObjectResp, err := blobberClient.RenameObject(grpcCtx, req)
+	if err != nil {
+		return nil, err
```

Comment:
Can we follow error wrapping over here? Something like this
```suggestion
		return nil, errors.Wrap(err, "failed to RenameObject")
```

Reply:
we actually, handle that error outside the client function, in here:
`	_, err = blobberClient.RenameObject(blobber.Baseurl, &blobbergrpc.RenameObjectRequest{
		Path:         req.remotefilepath,
		Allocation:   req.allocationTx,
		ConnectionId: req.connectionID,
		NewName:      req.newName,
	})
	if err != nil {
		Logger.Error("could not rename object from blobber -" + blobber.Baseurl + " - " + err.Error())
		return nil, err
	}
`
I think this pattern is followed across other client functions as well. I think wrapping the error on a level above should be good as well.

Reply:
Thats not wrapping the errors, that's just catching the errors. See this how wrapping works https://blog.golang.org/go1.13-errors , I have made sure that in blobber grpc handlers we wrap the errors whereever possible. Let's start doing this in the client too, we shouldn't repeat the mistakes whatever done in legacy